### DISCUSSION TITLE
Replacing Servers with P2P Connections

### DISCUSSION THREAD

**Discussion by Overload3910** [rustdesk/rustdesk#5598] [General]
Hey Guys,

are you familiar with Keet.io or Holepunch in general?

https://holepunch.to/

It's an easy and reliable way to connect in a peer 2 peer manner and replace servers that way. No open ports needed. 

There is a demo app out already, it's called keet. It's a Chat App supporting Video and Text. Please try it out and lmk if you think this is worth exploring. 

Thanks

**Comment by Mysticswiss** [comment]
Totally worth it , been using it extensively for some time . Fully P2P , no server , fast af , solid team . Can't recommend it enough .

**Comment by trymeouteh** [comment]
https://github.com/rustdesk/rustdesk/discussions/6865

**Comment by ryanamaral** [comment]
Hey folks, I just stumbled across this discussion. It might be worth looking at the project peerviewer.org as it uses Holepunch P2P connections.

**Comment by aabittner** [comment]
Hi there, the recent (yesterday?) outage of the central relay/bootstrap? (Vultr provider) infrastructure of Rustdesk was prompting me to write here: would it be possible to look into a more robust and smaller bus-factored way of bootstrapping and establishing a more peer to peer style of Rustdesk network of worldwide participating Rustdesk nodes. I was looking into the neat and stable @syncthing project (they are a go based project), maybe Rustdesk could take some ideas from them? I also wondered where and how the current client? certificates of Rustdesk become generated and stored and also cached? stored? eternally on some central Rustdesk back-ends and how this scales with growing Rustdesk user-base in general. Syncthing e.g. derives or creates hashed values from their locally generated and locally-only stored tls/ssl certificates of the clients and uses these hash sort of values (node IDs) to talk to relays and fellow nodes etc. maybe this would be a great idea? imho.

docs about their protocol types and specs: https://docs.syncthing.net/specs/index.html

maybe Rustdesk could also bring in some functionality of relays into the local clients themselves to offer and help with relaying via other fellow clients in a configurable way. thanks much.
